client/features/auth-pages/SignUpPage.tsx

```diff
@@ -8,21 +8,42 @@ import { useGithubAuthMutation } from '@hooks/useSignInWithGitHub';
 import { useCreateUserWithEmailAndPasswordMutation } from '@/hooks/useCreateUserWithEmailAndPassword';
 import type { SubmitHandler } from 'react-hook-form';
 import { useForm } from 'react-hook-form';
-import { z } from 'zod';
 import { zodResolver } from '@hookform/resolvers/zod';
 import useCreateUserProfileMutation from '@/hooks/useCreateUserProfileMutation';
 import SnackBarNotification from '@components/common/SnackBarNotification';
 import type { SnackbarSeverity } from '@/types/snackBar';
 import { useQueryClient } from '@tanstack/react-query';
 import useAuth from '@/hooks/useAuth';
+import { userPasswordSchema } from '@validators/validateUserPassword';
+import { userEmailSchema } from '@validators/validateUseremail';
+import { z } from 'zod';
+
+interface FormData {
+  email: string;
+  name: string;
+  password: string;
+  confirmPassword: string;
+}
+
+const passwordSchema = userPasswordSchema._def.schema as z.ZodObject<{
+  password: z.ZodString;
+  confirmPassword: z.ZodString;
+}>;
 
-const userSignUpSchema = z.object({
-  email: z.string().email({ message: 'Invalid email address' }),
-  password: z.string().min(1, { message: 'Please enter a password' }),
-  name: z.string().min(1, { message: 'Please enter your name' }),
+const baseSchema = z.object({
+  name: z.string().min(1, { message: `Name is Required` }),
+  email: userEmailSchema.shape.email,
 });
 
-type FormData = z.infer<typeof userSignUpSchema>;
+const combinedSchema = baseSchema.merge(passwordSchema);
```

Comment:
@tedgusek You're doing some great work here, and you're close to getting approval.

I have to call out these variable names - combinedSchema, baseSchema, and finalSchema all sound like names that imply ambiguous relationships to other variables. I'd like you to give some thought to whether all three are really necessary, and for all that are necessary please come up with variable names that communicate their purpose or origin more precisely.